src/generators/default/string/index.ts

```diff
@@ -14,23 +13,29 @@ function formatString(core: Core, def: ZodStringDef, value: string) {
 	const startsWith = core.utils.filterChecks(def.checks, 'startsWith')?.value;
 	const endsWith = core.utils.filterChecks(def.checks, 'endsWith')?.value;
 
+	if (length) {
+		max = length;
+	}
```

Comment:
Need to figure out what logic matches Zod exactly. I believe that if length is specified, the string HAS to be that length (ie min and max are the same). I'm not saying that's correct, but I want the fixture to be 100% in line with what Zod dictates.

Reply:
`length` does indeed mean that min and max have the same length.
As far as I know this implementation is in line with how zod works.
To be sure I also added tests for small and large strings.

I could be missing something though, what do you think that isn't in line with zod?

Reply:
It does get tricky when `startsWith` and/or `endsWith` is used but are larger than the `minLength`... But that's a zod issue I assume 😅

Reply:
I say we keep parity with whatever Zod is doing. Time to get our hands dirty and dig into the Zod source code? 🔍

Reply:
I think we're good to go though:

![image](https://github.com/timdeschryver/zod-fixture/assets/28659384/8cee025b-0a75-4f58-8a64-633c3cc06012)